app/controllers/entry_controller.rb

```diff
@@ -44,33 +42,25 @@ def create
   end
 
   def destroy
-    challenge = Challenge.find(params[:challenge])
-    entry = challenge.entries.find(params[:entry])
-
-    if challenge.owner?(current_user) || entry.owner?(current_user)
-      entry.destroy
-      challenge.save
-
-      flash[:notice] = "Deleted entry"
+    if @entry && (@challenge.owner?(current_user) || @entry.owner?(current_user))
```

Comment:
I realised here we are doing what the method load entry is already doing, so i added the callback for destroy method. I also thought that if we guard against `@entry` being nil then i think the rescue is not neccessary? what do you think?